README.md

```diff
@@ -1,22 +1,93 @@
-Optional parameter `ef` specifies the number of neighbors which are considered during index construction and search (corresponds `efConstruction` and `efSearch` parameters described in the article<sup>[[1]](#references)</sup>).
+## Enable the extension
 
-## Example of usage:
+Enable the `pg_embedding` extension in PostgreSQL by running the following `CREATE EXTENSION` statement.
 
+```sql
+CREATE EXTENSION embedding;
 ```
-create extension embedding;
-create table embeddings(id integer primary key, payload real[]);
-create index on embeddings using hnsw(payload) with (maxelements=1000000, dims=100, m=32);
-select id from embeddings order by payload <-> array[1.0, 2.0,...] limit 100;
+
+## Create a table for vector data
+
+To store vector data, create a table similar to the following:
+
+```sql
+CREATE TABLE documents(id INTEGER, embedding REAL[]);
 ```
 
+This statement generates a table named `documents` with a `embedding` column for storing vector data. (Your table and vector column names may differ.)
+
+## Insert data
+
+To insert vector data:
+
+```sql
+INSERT INTO documents(id, embedding) 
+VALUES 
+(1, '{1.1, 2.2, 3.3,...}'),
+(2, '{4.4, 5.5, 6.6,...}'),
+(3, '{7.7, 8.8, 9.9,...}');
+```
+
+## Create an HNSW index
+
+HNSW indexes are created in memory and built on demand.
+
+To create an HNSW index on your vector column, use a `CREATE INDEX` statement similar to the following:
+
+```sql
+CREATE INDEX ON documents USING hnsw(embedding) WITH (maxelements=1000000, dims=100, m=32);
+```
+
+## HNSW index options
+
+- `maxelements`: Defines the maximum number of elements indexed. This is a required parameter. An "element" refers to a data point (a vector) in the dataset, which is represented as a node in the HNSW graph. Typically, you would set this option to a value able to accommodate the number of rows in your in your dataset.
+- `dims`: Defines the number of dimensions in your vector data.  
+- `m`: Defines the maximum number of links (also referred to as "edges") created for each node during graph construction.
+- `efConstruction`: Defines the number of nearest neighbors considered during index construction. The default value is `32`. This is an optional parameter.
+- `efsearch`: Defines the number of nearest neighbors considered during index search. The default value is `32`. This is an optional parameter.
+
+## Query the indexed data
+
+To query the indexed data for nearest neighbors, use a query similar to this:
+
+```sql
+SELECT id FROM documents ORDER BY embedding <-> array[1.1, 2.2, 3.3,...] LIMIT 100;
+```
+
+where:
+
+- `SELECT id FROM documents` selects the `id` field from all records in the `documents` table.
+- `<->` is the PostgreSQL "distance between" operator. It calculates the Euclidean distance (L2) between the query vector and each row of the dataset.
```

Comment:
We define this operator in the extension, so I wouldn't say that it's Postgres'

https://github.com/neondatabase/pg_embedding/blob/2465f831ea1f8d49c1d74f8959adb7fc277d70cd/embedding--0.1.0.sql#L11-L14

Reply:
Fixed. Thank you.